tests/ga/test_remoteaccess_handler.py

```diff
@@ -0,0 +1,358 @@
+        remote_access.user_list.users[0].expiration = expiration
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertFalse("testAccount" in os_util.all_users)
+
+    def test_error_add_user(self):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        tstuser = "foobar"
+        expiration = datetime.utcnow() + timedelta(days=1)
+        pwd = "bad password"
+        rah.add_user(tstuser, pwd, expiration)
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertFalse(tstuser in os_util.all_users)  
+
+    def test_handle_remote_access_no_users(self):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_no_accounts.xml')
+        remote_access = RemoteAccess(data_str)
+        rah.remote_access = remote_access
+        rah.handle_remote_access()
+        os_util = rah.os_util
+        os_util.__class__ = MockOSUtil
+        self.assertEqual(0, os_util.all_users.keys().__len__())
+
+    def test_handle_remote_access_validate_jit_user_valid(self):
+        rah = RemoteAccessHandler()
+        result = rah.validate_jit_user("JIT Account")
+        self.assertTrue(result, "JIT account incorrectly identified.")
+
+    def test_handle_remote_access_validate_jit_user_invalid(self):
+        rah = RemoteAccessHandler()
+        result = rah.validate_jit_user("John Doe")
+        self.assertFalse(result, "JIT account incorrectly identified.")
+        
+    def test_handle_remote_access_validate_jit_user_None(self):
+        rah = RemoteAccessHandler()
+        result = rah.validate_jit_user(None)
+        self.assertFalse(result, "JIT account incorrectly identified.")
+
+    def test_handle_remote_access_validate_jit_user_blank(self):
+        rah = RemoteAccessHandler()
+        result = rah.validate_jit_user("")
+        self.assertFalse(result, "JIT account incorrectly identified.")
+
+    @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret',
+    return_value="]aPPEv}uNg1FPnl?")
+    def test_handle_remote_access_multiple_users(self, _):
+        rah = RemoteAccessHandler()
+        rah.os_util = MockOSUtil()
+        data_str = load_data('wire/remote_access_two_accounts.xml')
+        remote_access = RemoteAccess(data_str)
+        tstuser = remote_access.user_list.users[0].name
+        tstuser2 = remote_access.user_list.users[1].name
+        expiration_date = datetime.utcnow() + timedelta(days=1)
+        expiration_date2 = datetime.utcnow() + timedelta(days=5)
+        expiration = (expiration_date).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        expiration2 = (expiration_date2).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
+        remote_access.user_list.users[0].expiration = expiration
+        remote_access.user_list.users[1].expiration = expiration2
```

Comment:
All of this code duplication is a bad thing, even in test code. Consider folding it into a loop and hoisting the difference between the two instances (near as I can tell, just the expiration date computation) to a data structure paralleling the data_str you loaded from the XML test input. You do it in the ten_users case (below); do it here as well, please.

Reply:
changed.

---
In reply to: [194543012](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r194543012) [](ancestors = 194543012)